src/pages/CardLayout/Subscribe.vue

```diff
@@ -90,25 +98,22 @@
         this.page++;
         this.shiftPages();
       },
-      valid() {
-        return email_reg_exp.exec(this.email) !== null && this.acceptance === "on";
+      checkEmail() {
+        return this.email === '' ? undefined : validate(this.email, 'email');
       },
-      submit() {
-        if(!this.valid()) { return; }
-        const query = `
-        mutation {
-          subscribeMailingList(email: "${this.email}", source: "${this.source}") {
-            message
-          }
+      async submit() {
+        try {
+          await query(`
+            mutation sub($email: String!, $source: String) {
+              subscribeMailingList(email: $email, source: $source) {
+                message
+              }
+            }
+          `, { email: this.email, source: this.source });
+          this.next();
+        } catch (err) {
+          alert(err);
```

Comment:
`alert` ...yikes

Reply:
it made debugging easy lol

Reply:
i wanted console.error, but vue no like me xD (I need to get that modal going)

Reply:
i think we should de prioritize the modal for now though. hacker dashboard is most important right now!